File: SummaryRanges/SumRanges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function (nums) {
  let index = 0;
  let ranges = [];
  for (let num = 0; num < nums.length; num++) {
    if (num + 1 < nums.length && nums[num] + 1 == nums[num + 1]) {
      continue;
    } else if (index == num) {
      ranges.push(`${nums[index]}`);
    } else {
      ranges.push(`${nums[index]}->${nums[num]}`);
    }
    index = num + 1;
  }
  return ranges;
};

File: SummaryRanges/SumRanges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        ranges = []
        index = 0
        for num in range(len(nums)):
            if num + 1 < len(nums) and nums[num] + 1 == nums[num + 1]:
                continue
            elif index == num:
                ranges.append(f"{nums[index]}")
            else:
                ranges.append(f"{nums[index]}->{nums[num]}")
            index = num + 1
        return ranges